### PR TITLE
Added MoCoGAN

### PR DESCRIPTION
**Pull request by ayush12gupta**

### CODE REVIEW

File: generative_models/MoCoGAN_PyTorch/README.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+Params size (MB): 42.12
+Estimated Total Size (MB): 70.67
+----------------------------------------------------------------
+```
+
+## Results
+
+Some samples of the generated videos are as follows:
+
+ ![gif](https://github.com/ayush12gupta/model-zoo/blob/master/generative_models/MoCoGAN_PyTorch/assets/gif2.gif)   ![gif](https://github.com/ayush12gupta/model-zoo/blob/master/generative_models/MoCoGAN_PyTorch/assets/gif1.gif)   ![gif](https://github.com/ayush12gupta/model-zoo/blob/master/generative_models/MoCoGAN_PyTorch/assets/gif3.gif)
```
Comment (nirmal-suthar) [inline generative_models/MoCoGAN_PyTorch/README.md]: use local gif files (replace this with assets/...)

File: generative_models/MoCoGAN_PyTorch/main.py [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+    if cuda == True:
+        torch.cuda.set_device(0)
+
+    videos, current_path = preprocess(args)
+    num_vid = len(videos)
+    img_size = 96
+    channel = 3
+    d_E = 10
+    hidden_size = 100  
+    d_C = 50
```
Comment (nirmal-suthar) [inline generative_models/MoCoGAN_PyTorch/main.py]: add this to parser

File: generative_models/MoCoGAN_PyTorch/main.py [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+    # setup model #
+    dis_i = Image_Discriminator(channel)
+    dis_v = Video_Discriminator()
+    gen_i = Generator(channel, nz)
+    gru = GRU(d_E, hidden_size, gpu=cuda)
+    gru.initWeight()
+    
+    # setup optimizer #
+    lr = 0.0002
+    betas = (0.5, 0.999)
```
Comment (nirmal-suthar) [inline generative_models/MoCoGAN_PyTorch/main.py]: add to parser

File: generative_models/MoCoGAN_PyTorch/main.py [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+    optim_GRU = optim.Adam(gru.parameters(), lr=lr, betas=betas)
+
+    if cuda == True:
+        dis_i.cuda()
+        dis_v.cuda()
+        gen_i.cuda()
+        gru.cuda()
+        criterion.cuda()
+
+    trained_path = os.path.join(current_path, 'trained_models')
```
Comment (nirmal-suthar) [inline generative_models/MoCoGAN_PyTorch/main.py]: parser

File: generative_models/MoCoGAN_PyTorch/main.py [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+    plt.legend()
+    plt.savefig("plot.png")
+    #plt.show()
+
+if __name__ == '__main__':
+    main()
+
+
+
+
```
Comment (nirmal-suthar) [inline generative_models/MoCoGAN_PyTorch/main.py]: remove extra lines

File: generative_models/MoCoGAN_PyTorch/resize.py [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import os
```
Comment (nirmal-suthar) [inline generative_models/MoCoGAN_PyTorch/resize.py]: you can merge this file in utils.py

**Comment by ayush12gupta** [comment]
Done

File: generative_models/cGAN_PyTorch/README.md [hunk 7]
```diff
@@ -1,5 +1,7 @@
 # Pytorch Implementation of Conditional GAN 
+
```
Comment (nirmal-suthar) [inline generative_models/cGAN_PyTorch/README.md]: Remove cgan file from this pr

File: generative_models/MoCoGAN_PyTorch/README.md [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+$ python3 main.py --epochs 40000
+```
+> **_NOTE:_** on Colab Notebook use following command:
+```python
+!git clone link-to-repo
+%run main.py --epochs 40000
+```
+
+```
+usage: main.py [-h] [--batch-size BATCH_SIZE] [--epochs EPOCHS]
```
Comment (nirmal-suthar) [inline generative_models/MoCoGAN_PyTorch/README.md:17]: update this with new parser args

File: generative_models/cGAN_PyTorch/README.md (hunk 7 above)
Comment (ayush12gupta) [reply generative_models/cGAN_PyTorch/README.md]: Can you delete this pull request i made another one as I was not able to delete it

**Pull request closed without merging**